src/codegen/irgen/hooks.cpp

```diff
@@ -692,11 +692,14 @@ Box* execfile(Box* _fn, Box* globals, Box* locals) {
 
 Box* execMain(Box* boxedCode, Box* globals, Box* locals, PyCompilerFlags* flags) {
     if (PyTuple_Check(boxedCode)) {
-        RELEASE_ASSERT(!globals, "");
-        RELEASE_ASSERT(!locals, "");
+        if (globals || locals) {
```

Comment:
unfortunately I think this error messages are wrong :-(
I have seen that our whole exec/eval handling has quite a lot of smaller missing parts/differences with cpython and feels like it's not really finished. I would really prefer if we soon switch to a implementation which uses more of cpythons code. Would this be something which interests you? Otherwise I would take on the task